libgoal/transactions.go

```diff
@@ -706,42 +705,54 @@ func (c *Client) MakeUnsignedAssetDestroyTx(index uint64) (transactions.Transact
 func (c *Client) MakeUnsignedAssetConfigTx(creator string, index uint64, newManager *string, newReserve *string, newFreeze *string, newClawback *string) (transactions.Transaction, error) {
 	var tx transactions.Transaction
 	var err error
-	var ok bool
 
 	// If the creator was passed in blank, look up asset info by index
-	var params v1.AssetParams
+	var params generated.AssetParams
 	if creator == "" {
-		params, err = c.AssetInformation(index)
+		asset, err := c.AssetInformationV2(index)
 		if err != nil {
 			return tx, err
 		}
+		params = asset.Params
 	} else {
 		// Fetch the current state, to fill in as a template
-		current, err := c.AccountInformation(creator)
+		current, err := c.AccountInformationV2(creator, true)
```

Comment:
I was going to suggest this could be improved with `client.AccountAssetInformation`, but I don't think this side of the if statement needs to exist at all.

Reply:
I think I agree - I'll follow up on this in another PR